src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -625,21 +631,44 @@ private static TypeWithAnnotations DelegateReturnTypeWithAnnotations(MethodSymbo
                 parameterScopesBuilder.Add(scope);
                 parameterTypesBuilder.Add(ParameterTypeWithAnnotations(i));
             }
+
             var parameterRefKinds = parameterRefKindsBuilder.ToImmutableAndFree();
             var parameterScopes = parameterScopesBuilder.ToImmutableAndFree();
             var parameterTypes = parameterTypesBuilder.ToImmutableAndFree();
 
+            var lambdaSymbol = new LambdaSymbol(
+                Binder,
+                Binder.Compilation,
+                Binder.ContainingMemberOrLambda,
+                _unboundLambda,
+                parameterTypes,
+                parameterRefKinds,
+                refKind: default,
+                returnType: default);
+
+            ArrayBuilder<ConstantValue?>? parameterDefaultValueBuilder = null;
+            for (int i = 0; i < lambdaSymbol.Parameters.Length; i++)
+            {
+                var param = lambdaSymbol.Parameters[i];
+                var constVal = param.ExplicitDefaultConstantValue;
+                if (constVal != null)
+                {
+                    if (parameterDefaultValueBuilder == null)
+                    {
+                        parameterDefaultValueBuilder = ArrayBuilder<ConstantValue?>.GetInstance(ParameterCount);
+
+                        // front fill the array with nulls because all the parameters before this one must have been non-default.
+                        parameterDefaultValueBuilder.AddMany(null, i);
+                    }
+
+                    parameterDefaultValueBuilder.Add(param.ExplicitDefaultConstantValue);
+                }
+            }
+
+            var parameterDefaultValues = parameterDefaultValueBuilder?.ToImmutableAndFree() ?? default;
```

Comment:
>parameterDefaultValues [](http://example.com/codeflow?start=16&length=22)

Is this array used?

Reply:
> > parameterDefaultValues
> 
> Is this array used?

Not currently, but it will be in my next PR. I decided on this as a cut-off point for this PR